pkg/cache/tas_flavor.go

```diff
@@ -108,12 +108,21 @@ func (c *TASFlavorCache) snapshotForNodes(log logr.Logger, nodes []corev1.Node,
 	snapshot := newTASFlavorSnapshot(log, c.Levels)
 	nodeToDomain := make(map[string]utiltas.TopologyDomainID)
 	for _, node := range nodes {
-		levelValues := utiltas.LevelValues(c.Levels, node.Labels)
-		capacity := resources.NewRequests(node.Status.Allocatable)
-		domainID := utiltas.DomainID(levelValues)
-		snapshot.levelValuesPerDomain[domainID] = levelValues
-		snapshot.addCapacity(domainID, capacity)
-		nodeToDomain[node.Name] = domainID
+		ready := false
+		for _, cond := range node.Status.Conditions {
```

Comment:
Is there any reason not to use the field indexer to list nodes?

Reply:
The nodes are already cached / indexed in controller-runtime. 

We could have an index for virtual field like `metadata.ready` and use it to quickly filter only ready nodes, but this seems like overkill for now. I don't think a healthy system has more than 10% of non-ready nodes, so filtering out non-ready nodes should be very fast in code.

Reply:
> Doing this via indexer increases the code complexity

If we can introduce the indexer, we can get ride of removing this loop and the below if-else.
We just add the index selector to line 83. For sure, we need to have indexer function, but the indexer function is separately from TAS codes.

I think that the indexer solution is more simpler than the current impl based on removing the loop and if-else here.

Reply:
I see, so we would move the complexity from here to a separate indexer function. I'm ok with either approach.

Reply:
so the indexer on `ready nodes` and just loop over ready nodes to gather info about usage, correct?

Reply:
@tenzen-y given the PR is almost ready (not sure why the tests fail, PTAL @mszadkow ) would you be ok to merge it this way for 0.9.0 and cleanup for the future releases?